### PR TITLE
Adds 2018 Honda Fit support

### PR DESCRIPTION
**Pull request by energee**
https://github.com/commaai/opendbc/pull/100

### CODE REVIEW

**Comment by rbiasini** [comment]
@energee once conflicts are merged, is this good to go?

**Comment by rbiasini** [comment]
@energee would love to merge this in if somebody can confirm it works

**Comment by rbiasini** [comment]
Still testing needed?

**Comment by energee** [comment]
I think its close, there is a user who has it working and he may have a diff for this branch with some further changes.

**Comment by energee** [comment]
Side note, how is this PR a year old?

**Comment by rbiasini** [comment]
Time flies 🤷‍♂️

**Comment by AravindaDP** [comment]
Note: Variable names in interface.py are changed between 0.5 and 0.6. So it would need refactoring before merging.
I'm currently working with a fit owner to see if this works. Will update if we could get this working on 0.6.2

File: selfdrive/car/honda/interface.py
Comment (AravindaDP) [inline selfdrive/car/honda/interface.py]:
```suggestion
      ret.mass = 2644. * CV.LB_TO_KG + STD_CARGO_KG
```
It's now STD_CARGO_KG on 0.6.x

File: selfdrive/car/honda/interface.py
Comment (AravindaDP) [inline selfdrive/car/honda/interface.py]:
```suggestion
      tire_stiffness_factor = 0.444 # not optimized yet
      ret.lateralTuning.pid.kpV, ret.lateralTuning.pid.kiV = [[0.25], [0.06]]
```
Does tire_stiffness_factor is essential? How do we tune it?
Needed lat kpv and kiv reduced to get rid of steering vibration

File: selfdrive/car/honda/interface.py
Comment (AravindaDP) [inline selfdrive/car/honda/interface.py]:
```suggestion
      ret.longitudinalTuning.kpBP = [0., 5., 35.]
```

File: selfdrive/car/honda/interface.py
Comment (AravindaDP) [inline selfdrive/car/honda/interface.py]:
```suggestion
      ret.longitudinalTuning.kpV = [1.2, 0.8, 0.5]
```

File: selfdrive/car/honda/interface.py
Comment (AravindaDP) [inline selfdrive/car/honda/interface.py]:
```suggestion
      ret.longitudinalTuning.kiBP = [0., 35.]
```

File: selfdrive/car/honda/interface.py
Comment (AravindaDP) [inline selfdrive/car/honda/interface.py]:
```suggestion
      ret.longitudinalTuning.kiV = [0.18, 0.12]
```

**Review by AravindaDP — CHANGES_REQUESTED** [review]
I have worked with a Fit owner to get his vehicle working on 0.6.2 based on these changes.

cabana link: https://my.comma.ai/cabana/?route=99e3eaed7396619e%7C2019-08-12--13-56-01&max=19&url=https%3A%2F%2Fchffrprivate-vzn.azureedge.net%2Fchffrprivate3%2Fv2%2F99e3eaed7396619e%2Fda964fc10ba22af53289a955fb652b43_2019-08-12--13-56-01

File: selfdrive/car/honda/interface.py
Comment (energee) [reply selfdrive/car/honda/interface.py]: Tire stiffness affects oversteer/understeer. It should be set here even if its set to factor of 1.

**Comment by energee** [comment]
Can we test this, @AravindaDP ?

**Comment by AravindaDP** [comment]
@energee @rbiasini 
Apologies to bumping you again. Fit owner confirmed that this works on 0.6.3 as well subject to suggested changes I have marked above.
So would love if this gets merged in after accepting suggested changes. Let me know if any additional details are required.

**Comment by AravindaDP** [comment]
@energee @rbiasini 

Here is the cabana drive provided by fit user for this
https://my.comma.ai/cabana/?route=99e3eaed7396619e%7C2019-08-13--15-07-03&max=0&url=https%3A%2F%2Fchffrprivate-vzn.azureedge.net%2Fchffrprivate3%2Fv2%2F99e3eaed7396619e%2Fced4327cf0038699b7d1cbe59e5f8843_2019-08-13--15-07-03

**Comment by energee** [comment]
@vanillagorillaa ^

**Comment by energee** [comment]
Can we verify tire stiffness value on some turns? probably needs some adjustment

**Comment by AravindaDP** [comment]
@energee Yes that variable needs more attention probably.
What indications we should be looking for on turns? Low torque compared to stock on curves? overshooting or any other indications?
I guess there was some tuning doc earlier on openpilot community site but I can no longer find it
Thanks for your time

**Pull request merged**
